llvm/lib/Target/AIE/AIEEliminateDuplicatePHI.cpp

```diff
@@ -124,9 +124,14 @@ class AIEEliminateDuplicatePHI : public MachineFunctionPass {
     for (auto &MI : make_early_inc_range(MBB.phis())) {
       if (!isPointerTypePHI(MRI, MI))
         continue;
-      for (auto &PHI : make_early_inc_range(
-               make_range(std::next(MachineBasicBlock::instr_iterator(MI)),
-                          MachineBasicBlock::instr_iterator(*LastPHI)))) {
+
+      // Ensure that next MI is valid
+      auto NextMI = std::next(MachineBasicBlock::instr_iterator(MI));
+      if (NextMI == MBB.instr_end() || NextMI == LastPHI)
+        continue;
```

Comment:
nit: If we reached the the end, should we simply break here?

Reply:
yeah probably.. I'll change it.